File: Semester 4/Programming and Projection Enironments/mpp-proiect-csharp/Proiect_MPP/Proiect_MPP/domain/DTOAirportFlight.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Proiect_MPP.domain
{
    internal class DTOAirportFlight : Entity<int>
    {
        private string departureName;
        private string departureCity;
        private string destinationName;
        private string destinationCity;
        private DateOnly departureDate;
        private TimeOnly departureTime;
        private int freeSeats;

        public DTOAirportFlight(int id, string departureName, string departureCity, string destinationName, string destinationCity,
            DateOnly departureDate, TimeOnly departureTime, int freeSeats) : base(id)
        {
            this.departureName = departureName;
            this.departureCity = departureCity;
            this.destinationName = destinationName;
            this.destinationCity = destinationCity;
            this.departureDate = departureDate;
            this.departureTime = departureTime;
            this.freeSeats = freeSeats;
        }

        public string DepartureName { get => departureName; set => departureName = value; }
        public string DepartureCity { get => departureCity; set => departureCity = value; }
        public string DestinationName { get => destinationName; set => destinationName = value; }
        public string DestinationCity { get => destinationCity; set => destinationCity = value; }
        public DateOnly DepartureDate { get => departureDate; set => departureDate = value; }
        public TimeOnly DepartureTime { get => departureTime; set => departureTime = value; }
        public int FreeSeats { get => freeSeats; set => freeSeats = value; }
    }
}

File: Semester 4/Programming and Projection Enironments/mpp-proiect-java/Proiect_MPP/Thrift/src/main/java/root/proiect_mpp/domain/DTOAirportFlight.java
package root.proiect_mpp.domain;

import root.proiect_mpp.utils.Constants;

import java.time.LocalDate;
import java.time.LocalTime;

public class DTOAirportFlight implements Entity<Integer> {
    private int id;
    private String departureName;
    private String departureCityName;
    private String destinationName;
    private String destinationCityName;
    private LocalDate departureDate;
    private LocalTime departureTime;
    private int freeSeats;

    public DTOAirportFlight(int id, String departureCityName, String departureName, String destinationCityName,
                            String destinationName, LocalDate departureDate, LocalTime departureTime, int freeSeats) {
        this.id = id;
        this.departureName = departureName;
        this.departureCityName = departureCityName;
        this.destinationName = destinationName;
        this.destinationCityName = destinationCityName;
        this.departureDate = departureDate;
        this.departureTime = departureTime;
        this.freeSeats = freeSeats;
    }

    @Override
    public Integer getId() {
        return id;
    }

    @Override
    public void setId(Integer id) {
        this.id = id;
    }

    public int getFreeSeats() {
        return freeSeats;
    }

    public void setFreeSeats(int freeSeats) {
        this.freeSeats = freeSeats;
    }

    public String getDepartureName() {
        return departureName;
    }

    public void setDepartureName(String departureName) {
        this.departureName = departureName;
    }

    public String getDepartureCityName() {
        return departureCityName;
    }

    public void setDepartureCityName(String departureCityName) {
        this.departureCityName = departureCityName;
    }

    public String getDestinationName() {
        return destinationName;
    }

    public void setDestinationName(String destinationName) {
        this.destinationName = destinationName;
    }

    public String getDestinationCityName() {
        return destinationCityName;
    }

    public void setDestinationCityName(String destinationCityName) {
        this.destinationCityName = destinationCityName;
    }

    public String getDepartureDate() {
        return departureDate.format(Constants.DATE_FORMATTER);
    }

    public void setDepartureDate(LocalDate departureDate) {
        this.departureDate = departureDate;
    }

    public String getDepartureTime() {
        return departureTime.format(Constants.TIME_FORMATTER);
    }

    public void setDepartureTime(LocalTime departureTime) {
        this.departureTime = departureTime;
    }
}
